### PR TITLE
News rss

### PR DESCRIPTION
**Pull request by msotho**
This pull request improves the robustness and reliability of the RSS client by enhancing error handling, expanding content type support, and adding comprehensive tests to cover new edge cases. The changes ensure that the client gracefully handles various response types and errors, and that its behavior is well-tested.

**Error handling improvements:**
- Added logging and exception handling in `get_articles` to skip sources that raise a `NewspyException` and log a warning, preventing the entire process from failing due to one problematic source. [[1]](diffhunk://#diff-b83229a42affbd86d45f37d8b74c1209afbd369a24b707b1c94e7c07d0b9bd77R3) [[2]](diffhunk://#diff-b83229a42affbd86d45f37d8b74c1209afbd369a24b707b1c94e7c07d0b9bd77R12-R17) [[3]](diffhunk://#diff-b83229a42affbd86d45f37d8b74c1209afbd369a24b707b1c94e7c07d0b9bd77R45-R51)

**Content type handling:**
- Updated the HTTP client to recognize additional XML content types (`application/xml`, `text/xml`) when parsing RSS feeds, making the client more flexible with different server responses.

**Testing enhancements:**
- Added new tests to verify that `get_articles` correctly handles cases where the response is a string (e.g., HTML error page), `None`, or an HTTP error, ensuring it returns an empty list or skips sources as appropriate.
- Minor updates to existing tests to remove unnecessary type ignores and improve clarity. [[1]](diffhunk://#diff-9ca2dcd896753cd4a2e35df68a976524a0370889befcac9affa7aeb0bb82b21cL121-R231) [[2]](diffhunk://#diff-9ca2dcd896753cd4a2e35df68a976524a0370889befcac9affa7aeb0bb82b21cL231-R341)

### CODE REVIEW

File: newspy/rss/client.py [hunk 1]
```diff
@@ -38,9 +42,13 @@ def get_articles(
...
         for future in as_completed(futures):
-            resp_json = future.result()
+            try:
+                resp_json = future.result()
+            except NewspyException as exc:
+                logger.warning("Skipping a source due to error: %s", exc)
+                continue
 
-            if resp_json:
+            if resp_json and isinstance(resp_json, list):
```
[the remark this answers is not in the document]
Comment (msotho) [reply newspy/rss/client.py:51]: @gemini-code-assist I want to skip the logic as it does not make sense to process an empty list
